### PR TITLE
Update ethereum/tests to 9.0.4

### PR DESCRIPTION
**Pull request by carver**
### What was wrong?

Ethereum/tests hasn't been updated in a while. No London tests are available.

### How was it fixed?

Updated the tests, and fixed whatever came up.

### To-Do

[//]: # (Stay ahead of things, add list items here!)
- [x] Clean up commit history

[//]: # (For important changes that should go into the release notes please add a newsfragment file as explained here: https://github.com/ethereum/py-evm/blob/master/newsfragments/README.md)

[//]: # (See: https://py-evm.readthedocs.io/en/latest/contributing.html#pull-requests)
- [x] Add entry to the [release notes](https://github.com/ethereum/py-evm/blob/master/newsfragments/README.md)

#### Cute Animal Picture

put a cute animal picture link inside the parentheses

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -36,7 +36,7 @@ commands=
     native-blockchain-istanbul: pytest {posargs:tests/json-fixtures/blockchain/test_blockchain.py --fork Istanbul}
     native-blockchain-berlin: pytest {posargs:tests/json-fixtures/blockchain/test_blockchain.py --fork Berlin}
     native-blockchain-metropolis: pytest {posargs:tests/json-fixtures/blockchain/test_blockchain.py --fork Metropolis}
-    native-blockchain-transition: pytest {posargs:tests/json-fixtures/blockchain/test_blockchain.py -k BlockchainTests/TransitionTests}
+    native-blockchain-transition: pytest {posargs:tests/json-fixtures/blockchain/test_blockchain.py -k TransitionTests --tx '2*popen//execmodel=eventlet'}
```
Comment (fselmo) [inline tox.ini:39]: This is pretty nice. TIL

File: eth/vm/forks/london/blocks.py [hunk 2]
```diff
@@ -84,10 +84,10 @@ class LondonMiningHeader(rlp.Serializable, MiningHeaderAPI):
 
 
 class LondonBlockHeader(rlp.Serializable, BlockHeaderAPI):
-    fields = UNMINED_LONDON_HEADER_FIELDS + [
+    fields = UNMINED_LONDON_HEADER_FIELDS[:-1] + [
```
Comment (fselmo) [inline eth/vm/forks/london/blocks.py:87]: @carver This is unfortunate. What's the reason for the rigid order here?

**Review by fselmo — APPROVED** [review]
LGTM, just a quick q on the rigid ordering of the London block header fields that we missed on the first go-around

File: eth/vm/forks/london/blocks.py (hunk 2 above)
Comment (carver) [reply eth/vm/forks/london/blocks.py:87]: It just seems to be the unfortunate choice of the EIP, that the base fee is always at the end of list of fields.

**Pull request merged**